code/modules/mob/living/simple_animal/friendly/farm_animals.dm

```diff
@@ -326,7 +326,89 @@
 			qdel(src)
 	else
 		STOP_PROCESSING(SSobj, src)
+		
+/mob/living/simple_animal/sheep
+	name = "sheep"
+	desc = "It's so fluffy!"
+	icon_state = "sheep"
+	icon_living = "sheep"
+	icon_dead = "sheep_dead"
+	gender = FEMALE
+	mob_biotypes = list(MOB_ORGANIC, MOB_BEAST)
+	speak = list("baa?","baa","BAAAAAA")
+	speak_emote = list("bleats")
+	emote_hear = list("brays.")
+	emote_see = list("nibbles at the ground.")
+	speak_chance = 1
+	turns_per_move = 5
+	see_in_dark = 6
+	butcher_results = list(/obj/item/reagent_containers/food/snacks/meat/slab = 4)
+	response_help  = "pets"
+	response_disarm = "gently pushes aside"
+	response_harm   = "kicks"
+	attacktext = "kicks"
+	attack_sound = 'sound/weapons/punch1.ogg'
+	health = 40
+	maxHealth = 40
+	var/obj/item/udder/udder = null
+	var/shaved = FALSE
+	gold_core_spawnable = FRIENDLY_SPAWN
+	blood_volume = BLOOD_VOLUME_GENERIC
+
+	do_footstep = TRUE
+
+/mob/living/simple_animal/sheep/Initialize()
+	udder = new()
+	. = ..()
+
+/mob/living/simple_animal/sheep/Destroy()
+	qdel(udder)
+	udder = null
```

Comment:
qdel will handle this
```suggestion
```

Reply:
these other two suggestions are on all the milkable animals which i followed suit of, should I change them too?

Reply:
You can, but you don't have to